doc/source/cluster/kubernetes/examples/distributed-checkpointing-with-gcsfuse.md

```diff
@@ -0,0 +1,264 @@
+
+# Distributed Checkpointing with KubeRay and GCSFuse
+
+This example demonstrates how to orchestrate distributed checkpointing with KubeRay, using the GCSFuse CSI driver and Google Cloud Storage as the remote storage system.
+To illustrate the concepts, this guide uses the [Finetuning a Pytorch Image Classifier with Ray Train](https://docs.ray.io/en/latest/train/examples/pytorch/pytorch_resnet_finetune.html) example.
+
+## Why Distributed Checkpointing with GCSFuse?
+
+In large-scale, high-performance machine learning, distributed checkpointing is crucial for fault tolerance, ensuring that if a node fails during training, the process can be resumed from the latest saved checkpoint instead of starting from scratch.
+While it's possible to directly reference remote storage paths (e.g., `gs://my-checkpoint-bucket`), using Google Cloud Storage FUSE (GCSFuse) offers distinct advantages for distributed applications. GCSFuse allows you to mount Cloud Storage buckets like local file systems, making checkpoint management more intuitive for distributed applications that rely on these semantics. Furthermore, GCSFuse is designed for high-performance workloads, delivering the performance and scalability needed for distributed checkpointing of large models.
+
+[Distributed checkpointing](https://docs.ray.io/en/latest/train/user-guides/checkpoints.html), in combination with [GCSFuse](https://cloud.google.com/storage/docs/gcs-fuse), allows for larger-scale model training with increased availability and efficiency.
+
+## Create a Kubernetes cluster on GKE
+
+Create a GKE cluster with the [GCSFuse CSI driver](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/cloud-storage-fuse-csi-driver) and [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) enabled, as well as a GPU node pool with 4 L4 GPUs:
+```
+export PROJECT_ID=<your project id>
+gcloud container clusters create kuberay-with-gcsfuse \
+    --addons GcsFuseCsiDriver \
+    --cluster-version=1.29.3-gke.1093000 \
+    --location=us-east4-c \
+    --machine-type=g2-standard-8 \
+    --num-nodes=4 \
+    --accelerator type=nvidia-l4,count=1,gpu-driver-version=latest \
+    --workload-pool=${PROJECT_ID}.svc.id.goog
+```
+
+## Install the KubeRay operator
+
+Follow [Deploy a KubeRay operator](kuberay-operator-deploy) to install the latest stable KubeRay operator from the Helm repository.
+The KubeRay operator Pod must be on the CPU node if you set up the taint for the GPU node pool correctly.
+
+## Configuring the GCS Bucket
+
+Create a GCS bucket that will be used as the remote filesystem.
+```
+gcloud storage buckets create gs://<your bucket name>
+```
+
+Create a Kubernetes ServiceAccount that will be used to grant the RayCluster access to mount the GCS bucket:
+```
+kubectl create serviceaccount pytorch-distributed-training
+```
+
+Bind the `roles/storage.objectUser` role to the Kubernetes service account and bucket IAM policy:
```

Comment:
The output of the `gcloud storage buckets add-iam-policy-binding` command should tell you if the binding was successful. Otherwise you won't know until you try to access the bucket with the service account.